File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    answer = 'Yes'
    p1, p2 = 0, 0   # cards1,2의 각 인덱스를 가리키는 포인터
    
    for i in range(len(goal)):
        if p1 < len(cards1) and cards1[p1] == goal[i]:
            p1 += 1
        elif p2 < len(cards2) and cards2[p2] == goal[i]:
            p2 += 1
        else:
            answer = 'No'
            break
    return answer

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.swift
import Foundation

func solution(_ cards1:[String], _ cards2:[String], _ goal:[String]) -> String {
    var cards1 = cards1
    var cards2 = cards2
    
    for target in goal {
        if !cards1.isEmpty && cards1[0] == target {
            cards1.removeFirst()
        } else if !cards2.isEmpty && cards2[0] == target {
            cards2.removeFirst()
        } else {
            return "No"
        }        
    }
    
    return "Yes"
}